### PR TITLE
feat(builder): use find-cache-dir to support customizing cache directory location

### PR DESCRIPTION
**Pull request by kallevmercury**
Add support for customizing the cache directory using the `CACHE_DIR` environment variable via https://www.npmjs.com/package/find-cache-dir.

Resolves: https://github.com/rspack-contrib/storybook-rsbuild/issues/197

### CODE REVIEW

**Comment by stephenjudkins** [comment]
Thank you!

**Comment by fi3ework** [comment]
Thank you. General LGTM, I'm wondering do we need to document this out, and I found it's also not documented in Storybook's site.

**Comment by kallevmercury** [comment]
> Thank you. Generally LGTM, I'm wondering do we need to document this out, but I found it's also not documented in Storybook's site.

@fi3ework Thank you for checking this so quickly; let me know if there are any further changes you'd like to see for the contribution. I couldn't find the cache dir env variable documented on the main Storybook configuration either besides the Github issues and the Vite builder source code such as here: https://github.com/storybookjs/storybook/blob/abe4c8813017be23b96b3a4faee11344bd2d6163/code/builders/builder-vite/src/plugins/external-globals-plugin.ts#L55

File: packages/builder-rsbuild/src/preview/virtual-module-mapping.ts [hunk 1]
```diff
@@ -21,12 +22,12 @@ export const getVirtualModules = async (options: Options) => {
...
-    ? path.resolve(
-        process.cwd(),
-        // TODO: This is a hard code cache dir, as Rspack doesn't support virtual modules now.
-        // Remove this when Rspack supports virtual modules.
-        './node_modules/.cache/storybook/storybook-rsbuild-builder',
-      )
+    ? // TODO: This is a hard code cache dir, as Rspack doesn't support virtual modules now.
+      // Remove this when Rspack supports virtual modules.
+      (findCacheDirectory({
+        name: 'storybook-rsbuild-builder',
```
Comment (fi3ework) [inline packages/builder-rsbuild/src/preview/virtual-module-mapping.ts:28]: Is it on purpose to change the cache dir from `.cache/storybook/storybook-rsbuild-builder` to `.cache/storybook-rsbuild-builder` here?

**Comment by fi3ework** [comment]
> > Thank you. Generally LGTM, I'm wondering do we need to document this out, but I found it's also not documented in Storybook's site.
> 
> @fi3ework Thank you for checking this so quickly; let me know if there are any further changes you'd like to see for the contribution. I couldn't find the cache dir env variable documented on the main Storybook configuration either besides the Github issues and the Vite builder source code such as here: [storybookjs/storybook@`abe4c88`/code/builders/builder-vite/src/plugins/external-globals-plugin.ts#L55](https://github.com/storybookjs/storybook/blob/abe4c8813017be23b96b3a4faee11344bd2d6163/code/builders/builder-vite/src/plugins/external-globals-plugin.ts#L55)

Since the solution could be found in Storybook, I think we don't need to add documentation for now.

File: packages/builder-rsbuild/src/preview/virtual-module-mapping.ts (hunk 1 above)
Comment (fi3ework) [reply packages/builder-rsbuild/src/preview/virtual-module-mapping.ts:28]: I think it's fine to place it in either location. I can't recall why I initially put it inside `./storybook`—maybe it was just an unconscious choice. I'm glad to see we're now aligned with Storybook. Thank you.

**Review by fi3ework — APPROVED** [review]

File: packages/builder-rsbuild/src/preview/virtual-module-mapping.ts (hunk 1 above)
Comment (kallevmercury) [reply packages/builder-rsbuild/src/preview/virtual-module-mapping.ts:28]: Thank you!

**Pull request merged**